File: src/main/java/org/radarcns/consumer/realtime/condition/ConditionFactory.java
package org.radarcns.consumer.realtime.condition;

import org.radarcns.config.realtime.ConditionConfig;

public class ConditionFactory {

  public static Condition getConditionFor(ConditionConfig conditionConfig) {
    switch (conditionConfig.getName()) {
      case LocalJsonPathCondition.NAME:
        return new LocalJsonPathCondition(conditionConfig);
      default:
        throw new IllegalArgumentException(
            "The specified condition with name " + conditionConfig.getName() + " is not correct.");
    }
  }
}

File: src/main/java/org/radarcns/consumer/realtime/condition/ConditionFactory.kt
package org.radarcns.consumer.realtime.condition

import org.radarcns.config.realtime.ConditionConfig

/**
 * Factory class for [Condition]s. It instantiates conditions based on the configuration
 * provided for the given consumer.
 */
object ConditionFactory {
    @JvmStatic
    fun getConditionFor(conditionConfig: ConditionConfig): Condition {
        return when (conditionConfig.name) {
            LocalJsonPathCondition.NAME -> LocalJsonPathCondition(conditionConfig)
            else -> throw IllegalArgumentException(
                    "The specified condition with name " + conditionConfig.name + " is not correct.")
        }
    }
}